### ISSUE TITLE
akka-http HttpOriginRange.ALL sometimes is  replaced by the LanguageRange 

### ISSUE THREAD

**Issue by mumutu** [akka/akka#20403]
same as the title  though it was marked as a deprecated.. and it said it would a null value。but why it
finally turned into a LanguageRange

in my use case it is simple , to make the cors enable

```
 final HttpResponse httpResponse = HttpResponse.create()                                            .addHeader(AccessControlAllowOrigin.create(HttpOriginRange.ALL))

```

but sometimes the error was shown 

```
java.lang.ClassCastException: akka.http.scaladsl.model.headers.LanguageRange$$times$ cannot be cast to akka.http.scaladsl.model.headers.HttpOriginRange
        at akka.http.javadsl.model.headers.AccessControlAllowOrigin.create(AccessControlAllowOrigin.java:15)
```

And I see the code in LanguageRange, why init with the HttpOriginRange insteadof  jm.headers.LanguageRange

```

  JavaInitialization.initializeStaticFieldWith(
    `*`, classOf[jm.headers.HttpOriginRange].getField("ALL"))

```

I am new to scala && akka-http..so feel free to close this issue.

**Comment by johanandren** [comment]
Hi, the reason for the tricks with `JavaInitialization` can be found in this ticket and comment: https://github.com/akka/akka/issues/19162#issuecomment-195228014

I'm afraid the Javadoc on the deprecated field is pointing in the wrong direction, what you should be using is `HttpOriginRanges.ALL`

That it sometimes would be `akka.http.scaladsl.model.headers.LanguageRange$$times$` seems very strange, can you provide us with the full stacktrace rather than just the two first lines? Does it happen consistently, (and if so, can you provide a minimal repeater project for us)?

**Comment by mumutu** [comment]
I will try my best to reproduce it.  And I have read the comment on JavaInitialization  but still wonder why every scala object init thestatic java field with the same name except the LanguageRange
e.g In httpEntity
```
  JavaInitialization.initializeStaticFieldWith(
    Empty, classOf[jm.HttpEntity].getField("EMPTY"))
```

In RemoteAddress

```
  JavaInitialization.initializeStaticFieldWith(
    Unknown, classOf[jm.RemoteAddress].getField("UNKNOWN"))
```

**Comment by mumutu** [comment]
@johanandren  In worksheet it seem in LanguageRange the java init the wrong field

using following code we can simply get the exception
```
object Test extends App{
  println(model.headers.HttpOriginRange.`*`)
  LanguageRange
  println(AccessControlAllowOrigin.create(HttpOriginRange.ALL))

}

```

```
*
Exception in thread "main" java.lang.ClassCastException: akka.http.scaladsl.model.headers.LanguageRange$$times$ cannot be cast to akka.http.scaladsl.model.headers.HttpOriginRange
	at akka.http.javadsl.model.headers.AccessControlAllowOrigin.create(AccessControlAllowOrigin.java:15)
	at akka.http.impl.model.Test$.delayedEndpoint$akka$http$impl$model$Test$1(Test.scala:16)
	at akka.http.impl.model.Test$delayedInit$body.apply(Test.scala:12)
	at scala.Function0$class.apply$mcV$sp(Function0.scala:34)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
	at scala.App$class.main(App.scala:76)
	at akka.http.impl.model.Test$.main(Test.scala:12)
	at akka.http.impl.model.Test.main(Test.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)

```

**Comment by ktoso** [comment]
This is a bug, thanks for noticing – I'll fix and add regression tests

**Comment by ktoso** [comment]
Fixed, will be part of 2.4.5
